mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,394 @@
+
+Getting started with Tag Helpers
+------------------------------------
+
+This tutorial provides an introduction to programming Tag Helpers. :doc:`intro` describes the benefits that Tag Helpers provide.
+
+A tag helper is any class that implements the ``ITagHelper`` interface. However, when you author a tag helper, you generally derive from ``TagHelper``, doing so gives you access to the ``Process`` method. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial. 
+
+#. Create a new ASP.NET MVC 6 project called **AuthoringTagHelpers**. You won't need authentication for this project.
+#. Create a folder to hold the Tag Helpers called *TagHelpers*. The *TagHelpers* folder is *not* required, but it is a reasonable convention. Now let's get started writing some simple tag helpers.
+
+Starting the email Tag Helper
+--------------------------------
+
+In this section we will write a tag helper that updates an email tag. For example:
+
+``<email>Support</email>``
+
+The server will use our email tag helper to convert that markup into the following:
+
+.. code-block:: HTML
+
+	<a href="mailto:Support@contoso.com">Support@contoso.com</a>
+
+That is, an anchor tag that makes this an email link. You might want to do this if you are writing a blog engine and need it to send email for marketing, support, and other contacts, all to the same domain.
+
+#. Add the following ``EmailTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/TagHelpers/z1EmailTagHelperCopy.cs
+   :language: c#
+   
+**Notes:** 
+
+- Tag helpers use a naming convention that targets elements of the root class name (minus the *TagHelper* portion of the class name). In this example, the root name of **Email**\TagHelper is *email*, so the ``<email>`` tag will be targeted. This naming convention should work for most tag helpers, later on I'll show how to override it.
+
+- The ``EmailTagHelper`` class derives from ``TagHelper``. The ``TagHelper`` class provides the rich methods and properties we will examine in this tutorial.
+- The  overridden ``Process`` method controls what the tag helper does when executed. The ``TagHelper`` class also provides an asynchronous version (``ProcessAsync``) with the same parameters.
+- The context parameter to ``Process`` (and ``ProcessAsync``) contains information associated with the execution of the current HTML tag.
+- The output parameter to ``Process`` (and ``ProcessAsync``) contains a stateful HTML element representative of the original source used to generate an HTML tag and content.
+- Our class name has a suffix of **TagHelper**, which is *not* required, but it's considered a best practice convention. You could declare the class as:
+
+.. code-block:: c#
+
+	public class Email : TagHelper
+
+And even without the ``[TargetElement]`` attribute, ``<email>`` tags will be targeted.
```

Comment:
We don't know about 'TargetElementAttribute`s yet.

`And even without the **TagHelper** suffix, ``<email>`` tags will be targeted.`